cscs-checks/microbenchmarks/gpu/gpu_burn/src/gpu_burn.cu

```diff
@@ -39,553 +42,369 @@
     }
+    bt->start_timer();
 
     // The actual work
-    /*int iters = 0;
-    unsigned long long int errors = 0;*/
-    try {
-        while (true) {
-            our->compute();
-            our->compare();
-            /*errors += our->getErrors();
-            iters++;*/
-            int ops = our->getIters();
-            write(writeFd, &ops, sizeof(int));
-            ops = our->getErrors();
-            write(writeFd, &ops, sizeof(int));
-        }
-    }
-    catch (std::string e) {
-    fprintf(stderr, "Failure during compute: %s\n", e.c_str());
-    int ops = -1;
-    // Signalling that we failed
-    write(writeFd, &ops, sizeof(int));
-    write(writeFd, &ops, sizeof(int));
-    exit(111);
-    }
-}
+    while (burn) {
+        test.compute();
+        test.compare();
 
-void updateTemps(std::vector<int> *temps) {
-    const int readSize = 10240;
-    static int gpuIter = 0;
-    char data[readSize+1];
-    unsigned int device_count, i;
-    int curPos = 0;
-    nvmlInit();
-    nvmlDeviceGetCount(&device_count);
-    for (i = 0; i < device_count; i++) {
-        nvmlDevice_t device;
-        nvmlDeviceGetHandleByIndex(i, &device);
-        unsigned int tempValue;
-        nvmlDeviceGetTemperature ( device, NVML_TEMPERATURE_GPU, &tempValue);
-        temps->at(gpuIter) = (int)tempValue;
-        gpuIter = (gpuIter+1)%(temps->size());
+        // Update the results
+        bt->log(test.getErrors());
     }
 }
 
-void listenClients(std::vector<int> clientFd, std::vector<pid_t> clientPid, int runTime) {
-    fd_set waitHandles;
 
-    // pid_t tempPid;
+template<class T> void launch(int duration)
+{
+    // Initializing A and B with random data
+    T *A = (T*) malloc(sizeof(T)*SIZE*SIZE);
+    T *B = (T*) malloc(sizeof(T)*SIZE*SIZE);
+    srand(10);
+    for (size_t i = 0; i < SIZE*SIZE; ++i) {
+        A[i] = (T)((double)(rand()%1000000)/100000.0);
+        B[i] = (T)((double)(rand()%1000000)/100000.0);
+    }
+
     char hostname[256];
     hostname[255]='\0';
     gethostname(hostname,255);
-    int maxHandle = 0;
-    FD_ZERO(&waitHandles);
 
-    for (size_t i = 0; i < clientFd.size(); ++i) {
-        if (clientFd.at(i) > maxHandle)
-            maxHandle = clientFd.at(i);
-        FD_SET(clientFd.at(i), &waitHandles);
-    }
-
-    std::vector<int> clientTemp;
-    std::vector<int> clientErrors;
-    std::vector<int> clientCalcs;
-    std::vector<struct timespec> clientUpdateTime;
-    std::vector<float> clientGflops;
-    std::vector<bool> clientFaulty;
-
-    time_t startTime = time(0);
-
-    for (size_t i = 0; i < clientFd.size(); ++i) {
-        clientTemp.push_back(0);
-        clientErrors.push_back(0);
-        clientCalcs.push_back(0);
-        struct timespec thisTime;
-        clock_gettime(CLOCK_REALTIME, &thisTime);
-        clientUpdateTime.push_back(thisTime);
-        clientGflops.push_back(0.0f);
-        clientFaulty.push_back(false);
-    }
+    // Initialise the SMI
+    Smi * smi_handle = new Smi();
```

Comment:
Couldn't you simply have `Smi smi_handle` here?

Reply:
Yes - forgot to clean that up.